### PR TITLE
Update Nimish's Endterm GSoC Blog Post

### PR DESCRIPTION
**Pull request by nimishbongale**

### CODE REVIEW

File: _gsocblogs/2022/blog_Rucio_NimishBongale.md [hunk 1]
```diff
@@ -1,127 +1,325 @@
...
-intro: Let’s have a look at my progress, learning, and overall experience during the GSoC 2022 period, shall we?
+intro: The Rucio WebUI 2.0 we've all been looking forward to is finally here...
 ---
 
-## Project Synopsis
+## Project Recap
 
 ### Mentors: Mayank Sharma, Martin B, Mario Lassnig
+
 ### Project Proposal link: [https://summerofcode.withgoogle.com/programs/2022/projects/1ps6wuOz](https://summerofcode.withgoogle.com/programs/2022/projects/1ps6wuOz)
```
Comment (agheata) [inline _gsocblogs/2022/blog_Rucio_NimishBongale.md:17]: Please don't use explicit links, in particular in subtitles
```suggestion
[Project Proposal](https://summerofcode.withgoogle.com/programs/2022/projects/1ps6wuOz)
```

File: _gsocblogs/2022/blog_Rucio_NimishBongale.md [hunk 2]
```diff
@@ -1,127 +1,325 @@
...
+
 ### Live storybook deployment: [https://rucio.cern.ch/webui](https://rucio.cern.ch/webui)
```
Comment (agheata) [inline _gsocblogs/2022/blog_Rucio_NimishBongale.md:19]:
```suggestion
[Live storybook deployment](https://rucio.cern.ch/webui)
```

File: _gsocblogs/2022/blog_Rucio_NimishBongale.md [hunk 3]
```diff
@@ -1,127 +1,325 @@
...
-- Start replacing templating with the RESTful approach. Mock data and carry out a POC for authorization workflows.
+#### Login Page
 
----
+The Login page serves as the entry point to the entire application. It makes use of the custom built Form component, which supports field validation. It handles multiple auth mechanisms such as Username/Password, x509 ceritficate, OIDC & OIDC PKCE. A successful sign in leads us to the Home page, while the user is notified in case of an unsuccessful one.
 
-- Finalize the approach.
-- Keep the code ready for accepting storybook UI contents.
-- Completely replace templating with modern TSX code.
+![Login](https://raw.githubusercontent.com/nimishbongale/webui/feature-add-snapshots/public/Login.png)
```
Comment (agheata) [inline _gsocblogs/2022/blog_Rucio_NimishBongale.md:59]: All your figures are too big and need to be scaled to 100% width as below:
```suggestion
<img src="https://raw.githubusercontent.com/nimishbongale/webui/feature-add-snapshots/public/Login.png" alt="Login" width="100%">
```

File: _gsocblogs/2022/blog_Rucio_NimishBongale.md [hunk 4]
```diff
@@ -1,127 +1,325 @@
...
 
-- Get complete approval for the site contents.
-- Work on more UI components within StorybookJS.
-- Implement environment variable wrappers, fetch API calls.
+#### DID Metadata Page
 
----
+The DID Metadata page displays the different attributes associated to a DID, which a user can request to be shown by clicking an info button.
+
+![DID Metadata](https://raw.githubusercontent.com/nimishbongale/webui/feature-add-snapshots/public/DIDMetadata.png)
```
Comment (agheata) [inline _gsocblogs/2022/blog_Rucio_NimishBongale.md:71]: see above

File: _gsocblogs/2022/blog_Rucio_NimishBongale.md [hunk 5]
```diff
@@ -1,127 +1,325 @@
...
+
+#### Rule Metadata Page
+
+The Rule Metadata page displays the different attributes associated to a Rule, which a user can request to be shown by clicking on the scopename in blue.
+
+![Rule Metadata](https://raw.githubusercontent.com/nimishbongale/webui/feature-add-snapshots/public/RuleMetadata.png)
```
Comment (agheata) [inline _gsocblogs/2022/blog_Rucio_NimishBongale.md:77]: see above

File: _gsocblogs/2022/blog_Rucio_NimishBongale.md [hunk 6]
```diff
@@ -1,127 +1,325 @@
...
+
+#### List Rules Page
+
+As the name suggests, this page enlists all the rules associated to a given account, and provides options to take actions on them. This is largely enabled by the Table component and different filters provided at the top. Also present on the view are buttons to delete and create new rules, which redirects us to the Rule Definition page.
+
+![List Rules](https://raw.githubusercontent.com/nimishbongale/webui/feature-add-snapshots/public/ListRules.png)
```
Comment (agheata) [inline _gsocblogs/2022/blog_Rucio_NimishBongale.md:83]: see above

File: _gsocblogs/2022/blog_Rucio_NimishBongale.md [hunk 7]
```diff
@@ -1,127 +1,325 @@
...
+
+![Step 2](https://raw.githubusercontent.com/nimishbongale/webui/feature-add-snapshots/public/Step2.png)
+
+Step 3:
+
+![Step 3](https://raw.githubusercontent.com/nimishbongale/webui/feature-add-snapshots/public/Step3.png)
+
+Step 4:
+
+![Step 4](https://raw.githubusercontent.com/nimishbongale/webui/feature-add-snapshots/public/Step4.png)
```
Comment (agheata) [inline _gsocblogs/2022/blog_Rucio_NimishBongale.md:103]: see above for figures

File: _gsocblogs/2022/blog_Rucio_NimishBongale.md [hunk 8]
```diff
@@ -1,127 +1,325 @@
...
+
+#### Search Page
+
+On every page of the new WebUI, there is a Header containing the Rucio logo, on the click of which the collapsible drawer Navbar expands. To the right we have an option to log out of the UI. In the centre we have a Search field, which allows us to search for DIDs / RSEs from anywhere within the application. This redirects us to the Search page, containing tables to show the relevant information.
+
+![Search](https://raw.githubusercontent.com/nimishbongale/webui/feature-add-snapshots/public/Search.png)
```
Comment (agheata) [inline _gsocblogs/2022/blog_Rucio_NimishBongale.md:109]: see above

File: _gsocblogs/2022/blog_Rucio_NimishBongale.md [hunk 9]
```diff
@@ -1,127 +1,325 @@
...
+
+#### Error Boundary Page
+
+The Error Boundary page serves a good purpose, that in the advent of an error/exception, it gracefully shows the error in a separate page, which signals to the user that there is an issue with the application at the moment. Without Error boundaries, the webpage would throw up a plain white screen as React unmounts the entire DOM. The Error boundaries serve another purpose - a component backed try-catch allowing us to log any anomalies into a monitoring service.
+
+![Error Boundary](https://raw.githubusercontent.com/nimishbongale/webui/feature-add-snapshots/public/ErrorBoundary.png)
```
Comment (agheata) [inline _gsocblogs/2022/blog_Rucio_NimishBongale.md:115]: see above

File: _gsocblogs/2022/blog_Rucio_NimishBongale.md [hunk 10]
```diff
@@ -1,127 +1,325 @@
...
+
+The old WebUI had a direct DB dependency, which is a security debt. With the new WebUI 2.0, this direct DB dependency is completely removed, owing to its REST'ful architecture.
+
+### Key features
+
+#### Storybook Components (x15)
+
+Storybook is great for documenting UI components. The same was used to document and build close to 15 different components, ranging from input fields to cards to tables.
+
+![Storybook](https://raw.githubusercontent.com/nimishbongale/webui/feature-add-snapshots/public/Storybook.png)
```
Comment (agheata) [inline _gsocblogs/2022/blog_Rucio_NimishBongale.md:153]: see above

File: _gsocblogs/2022/blog_Rucio_NimishBongale.md [hunk 11]
```diff
@@ -1,127 +1,325 @@
...
+
+#### Accessibility
+
+a11y plays a pivotal role in determining if an application caters to audience of all forms. Using WAVE and other accessibility tools, there has been a solid attempt at improving the a11y score of the website since the day the first page was complete.
+
+### Relevant PRs
+
+- [Code Refactoring](https://github.com/rucio/webui/pull/98)
+
+#98 by [nimishbongale](https://github.com/rucio/webui/issues?q=is%3Apr+author%3Animishbongale) was merged 4 days ago
```
Comment (agheata) [inline _gsocblogs/2022/blog_Rucio_NimishBongale.md:221]: Please remove all the lines like this showing the status of the PR, just keep the links to the PRs

**Review by agheata — CHANGES_REQUESTED** [review]
@nimishbongale, please take the suggestions into account and ask your mentors to comment/approve your PR

**Comment by agheata** [comment]
Looks good to me, waiting for one of your mentors' OK before merging

**Comment by nimishbongale** [comment]
@maany @bari12

Please do review/approve the same, thank you! 😄

File: _gsocblogs/2022/blog_Rucio_NimishBongale.md [hunk 12]
```diff
@@ -1,127 +1,279 @@
...
+As the name suggests, this page enlists all the rules associated to a given account, and provides options to take actions on them. This is largely enabled by the Table component and different filters provided at the top. Also present on the view are buttons to delete and create new rules, which redirects us to the Rule Definition page.
 
-#51 opened 25 days ago by [nimishbongale](https://github.com/rucio/webui/issues?q=is%3Apr+is%3Aopen+author%3Animishbongale)
+<img src="https://raw.githubusercontent.com/nimishbongale/webui/feature-add-snapshots/public/ListRules.png" alt="List Rules" width="100%">
 
-- [Create REST API wrapper utility](https://github.com/rucio/webui/pull/50)
+#### Rule Definition / Create Rules Page
 
-#50 by [nimishbongale](https://github.com/rucio/webui/issues?q=is%3Apr+author%3Animishbongale) was merged 26 days ago• [Approved](https://github.com/rucio/webui/pull/50#partial-pull-merging)
+Central to the functionality of the Rucio WebUI is the ability to create rules with just a few clicks.Once a rule is created, the rule automatically reflects in the List Rules page, owing to the implementation of the global store using the React Context API.
```
Comment (maany) [inline _gsocblogs/2022/blog_Rucio_NimishBongale.md:87]: We should also credit Valentina for laying the foundations of the error and rule workflow pages and the corresponding storybook components ;). You must mention here the work you have done on top to refactor it to the new architecture and clean up the UI to make it fit more with the rest of the components

File: _gsocblogs/2022/blog_Rucio_NimishBongale.md [hunk 13]
```diff
@@ -1,127 +1,279 @@
...
----
+In total 8 pages were created as a part of this endeavour.
 
-- Finalize the approach.
-- Keep the code ready for accepting storybook UI contents.
-- Completely replace templating with modern TSX code.
+#### Login Page
 
----
+The Login page serves as the entry point to the entire application. It makes use of the custom built Form component, which supports field validation. It handles multiple auth mechanisms such as Username/Password, x509 ceritficate, OIDC & OIDC PKCE. A successful sign in leads us to the Home page, while the user is notified in case of an unsuccessful one.
```
Comment (maany) [inline _gsocblogs/2022/blog_Rucio_NimishBongale.md:57]: You could also mention some of the challenges with OIDC, PKCE and x509 strategies somewhere in this document.

File: _gsocblogs/2022/blog_Rucio_NimishBongale.md [hunk 14]
```diff
@@ -1,127 +1,279 @@
...
 
-- [Initial Setup of Storybook, React App, CICD deploy to GitHub pages](https://github.com/rucio/webui/pull/16) [DevOps](https://github.com/rucio/webui/issues?q=is%3Apr+author%3Animishbongale+label%3ADevOps) [gsoc22](https://github.com/rucio/webui/issues?q=is%3Apr+author%3Animishbongale+label%3Agsoc22)
+### Key Takeaways
 
-#16 by [nimishbongale](https://github.com/rucio/webui/issues?q=is%3Apr+author%3Animishbongale) was merged on 10 Jun• [Approved](https://github.com/rucio/webui/pull/16#partial-pull-merging) [1.0.0](https://github.com/rucio/webui/milestone/1)
+- Having the right mentors are key to any project - a huge shoutout to @Mayank, @Martin & @Cedric! (special mention to @Valentina for parterning with me on this project!)
+- Creating an OSS project from scratch having a huge impact has been a dream of mine, and I think I very well might've done it!
+- Rethinking UI really requires you to step out of your developer shoes and don the product hat.
+- Being passionate about something really drives you to do it consistently.
+- While the GSoC period ends here, contributing to OSS should continue!
```
Comment (maany) [inline _gsocblogs/2022/blog_Rucio_NimishBongale.md:277]: Maybe you could also add a section to summarize the pending tasks and what could be improved :)

**Review by maany — CHANGES_REQUESTED** [review]
Well done @nimishbongale. It is a pretty extensive and exhaustive report. I have left some minor comments

**Review by maany — APPROVED** [review]
Looks good, could you use her github handle [elizabethh777](https://github.com/elizabethh777)

**Comment by nimishbongale** [comment]
> Looks good, could you use her github handle [elizabethh777](https://github.com/elizabethh777)

Done!

**Review by maany — APPROVED** [review]
Looks good to go!

**Review by agheata — APPROVED** [review]

**Pull request merged**